File: AbsenceAppV1/src/AbsenceApp.Data/Models/UserProfile.cs
/*
===============================================================================
 File        : UserProfile.cs
 Namespace   : AbsenceApp.Data.Models
 Author      : Michael
 Version     : 1.0.0
 Created     : 2026-03-15
 Updated     : 2026-03-15
-------------------------------------------------------------------------------
 Purpose     : EF Core entity for the user_profiles table.
               Holds extended personal and organisational profile data for a user, including
               job title, department, school, and localisation preferences.
-------------------------------------------------------------------------------
 Changes     :
   - 1.0.0  2026-03-15  Initial creation.
-------------------------------------------------------------------------------
 Notes       :
   - No navigation properties; FK integrity is enforced at the database layer.
   - All required string properties use = default! to satisfy the nullable compiler.
===============================================================================
*/
namespace AbsenceApp.Data.Models;

public class UserProfile
{
    public long Id { get; set; }
    public long UserId { get; set; }
    public string FirstName { get; set; } = default!;
    public string LastName { get; set; } = default!;
    public string? PreferredName { get; set; }
    public string Title { get; set; } = default!;
    public DateOnly DateOfBirth { get; set; }
    public string? ProfilePictureUrl { get; set; }
    public string? Bio { get; set; }
    public string? Gender { get; set; }
    public string Timezone { get; set; } = default!;
    public string LanguageCode { get; set; } = default!;
    public long DepartmentId { get; set; }
    public long JobTitleId { get; set; }
    public long SchoolId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: Ki-Admin/app/Models/UserProfile.php
<?php
/**
 * =========================================================
 * File: UserProfile.php
 * Project: Ki-Admin - v1.0.0
 * 
 * Description:
 *   Eloquent model for user_profiles table.
 *   Handles extended user data including:
 *     - Personal and contact details
 *     - Profile picture and biography
 *     - Job grouping and organizational mappings
 * 
 * Updated: 2025-08-11
 * Author : Michael Battle
 * =========================================================
 */

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

use App\Models\User;
use App\Models\JobGroup;
use App\Models\Department;
use App\Models\JobTitle;
use App\Models\School;

class UserProfile extends Model
{
    /* =====================================================
     * Section: Table & Timestamps
     * Description: Table name and timestamp flags
     * ===================================================== */
    protected $table = 'user_profiles';
    public $timestamps = true;

    /* =====================================================
     * Section: Mass Assignment
     * Description: Fields that can be bulk-assigned
     * ===================================================== */
    protected $fillable = [
        'user_id',
        'first_name',
        'last_name',
        'preferred_name',
        'title',
        'date_of_birth',
        'profile_picture_url',
        'biography',
        'phone_number',
        'job_group_id',
        'gender',
        'timezone',
        'language',
        'department_id',
        'job_title_id',
        'school_id',
    ];

    /* =====================================================
     * Section: User Relation
     * Description: Link to primary users table
     * ===================================================== */
    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class, 'user_id');
    }

    /* =====================================================
     * Section: Job Group Relation
     * Description: Classification from job_group table
     * ===================================================== */
    public function jobGroup(): BelongsTo
    {
        return $this->belongsTo(JobGroup::class, 'job_group_id');
    }

    /* =====================================================
     * Section: Department Relation
     * Description: Organizational department link
     * ===================================================== */
    public function department(): BelongsTo
    {
        return $this->belongsTo(Department::class, 'department_id');
    }

    /* =====================================================
     * Section: Job Title Relation
     * Description: Standard job titles reference
     * ===================================================== */
    public function jobTitle(): BelongsTo
    {
        return $this->belongsTo(JobTitle::class, 'job_title_id');
    }

    /* =====================================================
     * Section: School Relation
     * Description: Link to associated school record
     * ===================================================== */
    public function school(): BelongsTo
    {
        return $this->belongsTo(School::class, 'school_id');
    }
}
